src/SSA_stepper.jl

```diff
@@ -141,10 +143,25 @@ function DiffEqBase.__init(jump_prob::JumpProblem,
                                cb,_saveat,save_everystep,save_end,cur_saveat,
                                opts,tstops,1,false,true)
     cb.initialize(cb,integrator.u,prob.tspan[1],integrator)
+    DiffEqBase.initialize!(opts.callback,integrator.u,prob.tspan[1],integrator)
     integrator
 end
 
-DiffEqBase.add_tstop!(integrator::SSAIntegrator,tstop) = integrator.tstop = tstop
+function DiffEqBase.add_tstop!(integrator::SSAIntegrator{<:Any,<:Any,<:Any,<:Any,<:Any,<:Any,<:Any,<:Any,<:Vector},tstop)
+    insert_index = searchsortedfirst(integrator.tstops, tstop)
```

Comment:
Looks good to me overall. Probably it won't matter for most applications, but couldn't you just search from `integrator.tstops_idx` on?

Reply:
Good idea!